File: C++/프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int start = 1, end = 1, sum = 1, answer = 0;
    while(start <= end) {
        if(sum == n) {
            answer++;
        }
        if(sum < n) {
            sum += ++end;
        } else {
            sum -= start++;
        }
    }
    return answer;
}

File: Python3/프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    sum = 1
    start, end = 1, 1
    while start <= end:
        if(sum == n):
            answer+=1
        if(sum < n):
            end += 1
            sum += end
        else:
            sum -= start
            start += 1
    return answer